archey/entries/terminal.py

```diff
@@ -1,11 +1,58 @@
 """Terminal detection class"""
 
 import os
+import re
 
 from archey.colors import Colors
 from archey.entry import Entry
 
 
+# We detect a terminal by using the following three constants in the order below:
+# First, we try using the value in the `TERM_PROGRAM` enviroment variable.
+# Then, we use `COLORTERM_DICT` to try matching a value with the `COLORTERM` environment variable.
+# Third, we use `TERM_DICT` to try matching a value with the $TERM environment variable.
+# Finally, we fall back to custom environment variables defined in `ENV_DICT`.
+# If none of the above tests find a value, we use whichever value was defined in `$TERM`.
+
+# All of the keys in `COLORTERM_DICT` and `TERM_DICT` are matched as regular expressions...
+#   (using `re.match`, i.e. attempting to match once at the beginning of the string), so be careful!
+
+
+# This dictionary contains values for the `COLORTERM` environment variable for terminal emulators
+#   which do not propogate any other usable environment variable.
+# If `COLORTERM` matches one of these values, a normalization is performed with its corresponding
+#   value (i.e. the respective terminal emulator).
+# If the variable does not match any keys in this dictionary, it is ignored.
+COLORTERM_DICT = {
+    r'kmscon': 'KMSCON',
+    r'rxvt': 'rxvt',
+}
+
+# This dictionary contains values for the `TERM` environment variable for terminal emulators
+#   which do not propogate any other usable environment variable.
+# If `TERM` matches one of these values, a normalization is performed with its corresponding
+#   value (i.e. the respective terminal emulator).
+# If the variable does not match any keys in this dictionary, it is ignored,
+#   UNLESS it does not begin with `xterm`, at which point its value is taken as the terminal in use.
+#   This behaviour can be overridden by specifying its exact match here.
+TERM_DICT = {
+    r'xterm-termite': 'Termite',
+}
```

Comment:
I have a feeling being able to use regular expressions to match here *could* be helpful in the future. Let me know whether you think so or not, because it *might* be more efficient not to use them?

Reply:
Yes, that was the reason I try to avoid them as much as possible.
In this case, it's _okay_, 'cause we try to match them against specific strings (two, actually), and not against each running processes names, so we keep this short algorithm `O(1)` :ok_hand: